ui/plugins/eu.esdihumboldt.hale.ui.common/src/eu/esdihumboldt/hale/ui/common/definition/viewer/OptionalPropertiesFilter.java

```diff
@@ -89,16 +91,20 @@ private boolean areChildrenOptional(EntityDefinition entityDef) {
 						.getConstraint(Cardinality.class);
 				if (cardinality.getMinOccurs() != 0)
 					return false;
+				else if (!areChildrenOptional(entityDef))
+					return false;
 			}
 			else if (def instanceof PropertyDefinition) {
 				Cardinality cardinality = ((PropertyDefinition) def)
 						.getConstraint(Cardinality.class);
 				if (cardinality.getMinOccurs() != 0)
 					return false;
```

Comment:
That is wrong. If minOccurs is not zero, you still have to check if it is Nillable, if it is zero you don't have to check.